src/components/TranslationValueRow/index.js

```diff
@@ -0,0 +1,70 @@
+// @flow
+import React, { Component } from 'react'
+import { type Locale } from '@types/Locale'
+import { type TranslationValue } from '@types/TranslationValue'
+import { Wrapper, ContentWrapper, LocaleWrapper, Textarea } from './styles'
+
+type Props = {
+  locale: Locale,
+  translationKeyValue: TranslationValue | null,
+  onBlur: string => any
+}
+
+type State = {
+  isFocused: boolean,
+  value: string
+}
+
+class TranslationValueRow extends Component<Props, State> {
+  state = {
+    isFocused: false,
+    value: ''
+  }
+
+  onChange = ({ target: { value } }: SyntheticInputEvent<HTMLInputElement>) => {
+    this.setState({ value })
+  }
+
+  toggleFocus = () => {
+    this.setState(state => ({ isFocused: !state.isFocused }))
+  }
+
+  onInputBlur = () => {
+    this.props.onBlur(this.state.value)
+    this.toggleFocus()
+  }
+
+  componentDidMount = () => {
+    const { translationKeyValue } = this.props
+
+    if (translationKeyValue) {
```

Comment:
why do you wait for `componentDidMount`? you can compute state from props in the constructor

```js
constructor (props) {
  super(props)
  this.state = { … }
}